src/components/core/PivotTable.tsx

```diff
@@ -243,6 +263,15 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
             if (agGridDataSourceUpdateNeeded) {
                 this.updateAGGridDataSource();
             }
+
+            const dataSourceChanged =
+                this.props.dataSource.getFingerprint() !== prevProps.dataSource.getFingerprint();
+            if (dataSourceChanged || totalsPropsChanged || totalsStateChanged) {
+                this.columnWidths = {};
```

Comment:
just wonder why columnWidth is not in state ? thats because double render?

Reply:
yes, that's the case